ionoscloud/resource_server.go

```diff
@@ -964,7 +972,11 @@ func resourceServerUpdate(ctx context.Context, d *schema.ResourceData, meta inte
 		}
 
 		// Wait, catching any errors
-		_, errState := cloudapi.GetStateChangeConf(meta, d, apiResponse.Header.Get("Location"), schema.TimeoutUpdate).WaitForStateContext(ctx)
+		loc, err := apiResponse.Location()
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		_, errState := cloudapi.GetStateChangeConf(meta, d, loc.String(), schema.TimeoutUpdate).WaitForStateContext(ctx)
 		if errState != nil {
 			diags := diag.FromErr(errState)
 			return diags
```

Comment:
this is a lot of repetition that could be refactored to it's own function

Reply:
which part? getting the location and checking state change?